homeassistant/components/discovergy/config_flow.py

```diff
@@ -70,8 +70,16 @@ async def async_step_reauth(
         self, entry_data: Mapping[str, Any]
     ) -> ConfigFlowResult:
         """Handle the initial step."""
-        self._existing_entry = await self.async_set_unique_id(self.context["unique_id"])
-        return await self._validate_and_save(entry_data, step_id="reauth")
+        self._existing_entry = self.hass.config_entries.async_get_entry(
+            self.context["entry_id"]
+        )
+        return await self.async_step_reauth_confirm()
```

Comment:
Small change in behavior...
Previously, the original credentials were tested, now the user is prompted for new credentials without attempting the old ones.
The tests imply that this was the desired behavior.